python/feathub/feature_tables/sources/datagen_source.py

```diff
@@ -144,33 +120,55 @@ def __init__(
                                      should be ordered by `timestamp_field`.
         """
         super().__init__(
-            name,
-            "datagen",
-            {},
+            name=name,
+            system_name="datagen",
+            properties={},
             keys=keys,
             timestamp_field=timestamp_field,
             timestamp_format=timestamp_format,
             schema=schema,
         )
 
-        self.data_gen_config = data_gen_config
+        self.number_of_rows = number_of_rows
+        self.rows_per_second = rows_per_second
+        self.field_configs = field_configs if field_configs is not None else {}
         self.max_out_of_orderness = max_out_of_orderness
 
         # TODO: Add validation of field type and field config.
-        for field, _ in data_gen_config.field_configs.items():
+        for field, _ in self.field_configs.items():
             if field not in schema.field_names:
                 raise FeathubException(f"Field {field} is not in the schema.")
 
         for field in schema.field_names:
-            if field not in data_gen_config.field_configs:
-                data_gen_config.field_configs[field] = default_field_config
+            if field not in self.field_configs:
+                self.field_configs[field] = RandomField()
+
+    def is_bounded(self) -> bool:
+        if self.number_of_rows is not None:
+            return True
+
+        for field_config in self.field_configs.values():
+            if isinstance(field_config, SequenceField):
+                return True
+
+        return False
+
+    def get_bounded_view(self) -> TableDescriptor:
+        if self.is_bounded():
+            return self
+
+        data_gen_source = deepcopy(self)
+        data_gen_source.number_of_rows = DEFAULT_BOUNDED_NUMBER_OF_ROWS
+        return data_gen_source
 
     def to_json(self) -> Dict:
         return {
             "type": "DataGenSource",
             "name": self.name,
             "schema": self.schema,
-            "data_gen_config": self.data_gen_config.to_json(),
+            "rows_per_second": self.rows_per_second,
+            "number_of_rows": self.number_of_rows,
+            "field_configs": {k: v.to_json() for k, v in self.field_configs.items()},
```

Comment:
Should we explicitly handle the case where `field_configs` is `None`?

Reply:
This self.data_gen_config should not be null. We assign an empty dict in the __init__ method if the user doesn't give one.